src/common/utils/resolve_reference.py

```diff
@@ -46,20 +46,19 @@ def reference_to_reference_items(reference: str, items=None, prev_deliminator=No
 
     if "$" in content:  # By id
         items.append(IdItem(content[1:]))
-    elif prev_deliminator == "/":
-        if len(items) > 0:  # By package
-            items.append(AttributeItem("content"))
-            items.append(QueryItem(query=f"name={content}"))
-        else:  # By root package
-            items.append(QueryItem(query=f"name={content},isRoot=True"))
+    elif prev_deliminator == "/" and len(items) == 0:  # By root package
+        items.append(QueryItem(query=f"name={content},isRoot=True"))
+    elif prev_deliminator == "/":  # By package
+        items.append(AttributeItem("content"))
+        items.append(QueryItem(query=f"name={content}"))
     elif prev_deliminator == "(" and deliminator == ")":  # By query
         items.append(QueryItem(query=content))
     elif prev_deliminator == "[" and deliminator == "]":  # By list index
         items.append(AttributeItem(f"[{content}]"))
     elif prev_deliminator == ".":  # By attribute
         items.append(AttributeItem(content))
-    else:  # Fallback to support the old reference format
-        items.append(IdItem(content))
+    else:
+        raise Exception("Not supported reference format")
```

Comment:
use ValidationException instead?
I believe that will give better error message for dm-cli / frontend.

Can also include  the reference as part of error message:
```python
raise Exception(f"Not supported reference format: {reference}")
```